doc/scaling_spire.md

```diff
@@ -0,0 +1,48 @@
+# Scalability
+
+A SPIRE deployment is composed of a number one or more SPIRE Servers that share a replicated datastore, and one or more SPIRE Agents. With only one SPIRE server per SPIRE deployment, the memory and CPU usage tends to grow proportionally with the number of workloads in the deployment. To support very large numbers of workloads and agents within a given trust domain (tens of thousands or hundreds of thousands of nodes), the number of SPIRE Servers can be scaled horizontally. With multiple servers, the load is distributed. In addition to capacity, multiple SPIRE Servers can be configured for high availability.
```

Comment:
There is one spot where we say "SPIRE server" instead of "SPIRE Server". Suggest being consistent one way or the other.